---
layout: post
title:  "Surface Roughness Induces Intraspecific Ocean Acidification Resistance"
author: "Patrick M Kiel"
date: '2024-05-11'
categories: [research]
description: "Online companion to Patrick Kiel's poster presented at the Microscale Ocean Biophysics Conference 7.0 on Heron Island, Australia."
output:
  md_document:
    variant: gfm
    preserve_yaml: TRUE
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, 
  encoding = encoding, 
  output_file = paste0('2024-05-11', "-",
                        gsub(pattern = "\\.Rmd$",
                              "", basename(inputFile))
                        ,".md"), 
  output_dir = "../_posts") })
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

#load packages
library(tidyverse)
library(kableExtra)
library(readxl)
library(ggpubr)
library(rstatix)
library(wesanderson)

base_dir <- "C:/Users/pkiel/Github/notebook/" # i.e. where the jekyll blog is on the hard drive.
base_url <- "/notebook/" # subdirectory of patrickmkiel.com
fig_path <- "images/MOB7Conference/" #add a new folder if you want to group output into a single foder

# Set base directories
knitr::opts_knit$set(base.dir = base_dir, base.url = base_url)

# Set figure directories
knitr::opts_chunk$set(fig.path = fig_path,
                    cache.path = '../cache/',
                    fig.align = 'center', out.width = '90%',
                    dpi = 300,
                    message=FALSE, warning=FALSE,
                    cache = FALSE, results = "asis")


#set default color paletes
options(ggplot2.discrete.fill=wes_palette("Darjeeling1"))
options(ggplot2.discrete.colour=wes_palette("Darjeeling1"))
```


```{css}
img {
margin: 0 auto;
}

table {
    width: 90%;
    border: 0px solid #fff;
    border-collapse: collapse;
    overflow-x: auto;
    margin: 0 auto;
    display: block;
}
```

```{r FigCaptions, echo=FALSE}

#Figure and Table Caption Numbering
capTabNo = 1; capFigNo = 1;

#Function to add the Table Number
capTab = function(x){
    x = paste0("Table ",capTabNo,". ",x)
    capTabNo <<- capTabNo + 1
    cat(x)
}

#Function to add the Figure Number
capFig = function(x){
    x = paste0("<h5>Figure ",capFigNo,". ",x,"</h5>")
    capFigNo <<- capFigNo + 1
    cat(x)
}
```

<!-- Place all chunks, text, etc here as you would a normal RMarkdown document -->

# Overview

Thank you for checking out my poster at the <a href="https://www.microscalemeeting.org/about-the-meeting.html" target="_blank">Microscale Ocean Biophysics Meeting 7.0 at Heron Island, Australia.</a> My poster is titled, "Surface Roughness Induces Intraspecific Ocean Acidification Resistance". 

Coral restoration efforts are growing globally to increase gene flow, coral cover, and , all in an attempt to buy time while climate change and. The systematic investigation of these has highlighted the tremendous inter and intraspecific phyenotypic diversity naturally present within coral populations. While much of the community is focussed on understanding the molecular roles underlying these phenotype-genotype relationships, there is much to learn from the corals themselves. 


# Poster
<img src="`r paste0(base_url, fig_path, 'KielMOB7poster_final.png')`" alt="Poster presented at MOB7 in Heron Island, Australia"/>

# References
