---
layout: post
title:  "TGA Analysis of Coral Skeletons"
author: "Patrick M Kiel"
date: '2022-09-15'
categories: [research]
description: "Review of thermogravimetric analysis techniques for quanitative comparisons of coral skeletal properties."
output:
  md_document:
    variant: gfm
    preserve_yaml: TRUE
knit: (function(inputFile, encoding) {
  rmarkdown::render(inputFile, 
  encoding = encoding, 
  output_file = paste0('2022-09-15', "-",
                        gsub(pattern = "\\.Rmd$",
                              "", basename(inputFile))
                        ,".md"), 
  output_dir = "../_posts") })
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)

#load packages
library(tidyverse)
library(kableExtra)
library(readxl)
library(ggpubr)
library(rstatix)
library(seacarb)

base_dir <- "C:/Users/pkiel/Github/notebook/" # i.e. where the jekyll blog is on the hard drive.
base_url <- "/notebook/" # subdirectory of patrickmkiel.com
fig_path <- "images/tgaAnalysis/" #add a new folder if you want to group output into a single foder

# Set base directories
knitr::opts_knit$set(base.dir = base_dir, base.url = base_url)

# Set figure directories
knitr::opts_chunk$set(fig.path = fig_path,
                    cache.path = '../cache/',
                    fig.align = 'center', out.width = '90%',
                    message=FALSE, warning=FALSE,
                    cache = FALSE, results = "asis")
```

```{js}

window.onload = function() {
    //query string in the password
    const urlParams = new URLSearchParams(window.location.search);
    const pass = urlParams.get('pass')
    document.getElementById("password").value = pass;
};

function verify() {
  <!-- set the password here -->
  if (document.getElementById('password').value === 'tga') {
    document.getElementById('HIDDENDIV').classList.remove("hidden"); 
    document.getElementById('credentials').classList.add("hidden"); // Hide the div containing the credentials
  } else {
    alert('Invalid Password! You cannot view this content.');
    password.setSelectionRange(0, password.value.length);
  }
  return false;
}
```

```{css}
/*Change content Display */
.hidden {
  display: none;
}
img {
    max-width: 90%;
    margin: 0 auto;
}
```

```{r FigCaptions, echo=FALSE}

#Figure and Table Caption Numbering
capTabNo = 1; capFigNo = 1;

#Function to add the Table Number
capTab = function(x){
    x = paste0("Table ",capTabNo,". ",x)
    capTabNo <<- capTabNo + 1
    cat(x)
}

#Function to add the Figure Number
capFig = function(x){
    x = paste0("<h5>Figure ",capFigNo,". ",x,"</h5>")
    capFigNo <<- capFigNo + 1
    cat(x)
}
```

<!-- The password box -->
<div id="credentials">
  <input type="text" id="password" onkeydown="if (event.keyCode == 13) verify()" />
  <br/>
  <input id="button" type="button" value="Show Content" onclick="verify()" />
</div>

<!-- The content we want to show after password -->
<div id="HIDDENDIV" class="hidden" markdown="1">

<!-- Place all chunks, text, etc here as you would a normal RMarkdown document -->

# Overview

Thermogravimetric analysis (TGA) is a quantitative materials technique that linearly heats a sample and simultaneously measures its mass as the sample degrades. By looking at the percent decomposition over discrete regions, we can characterize the coral sample's fractional constituents. The heating occurs in an inert environment (N<sub>2</sub> gas) to avoid sample oxidation. 

For analysis, we can use the first derivative of the TGA curve (called the DTG curve) to identify the start, peak, and end temperatures for each respective region. Then, we can integrate between the start and end temperatures to calculate the percent degradation. For example, an aragonite coral skeleton may have two peaks: decomposition of the intracrystaline organics at approximately  and the decomposition of CaCO<sub>3</sub> to CaO and CO<sub>2</sub> at approximately 600-800°C (Fig 1.).

```{r example-TGA}
  exampleTGA = readLines('~/Grad School/ULink 2022/TGA/allysonGenotypes/pmk_genos_149.txt')
  filteredLines = head(tail(exampleTGA,-10),-2)
  dat = read.csv(textConnection(filteredLines), header = FALSE, stringsAsFactors = FALSE,
                 sep = "\t")
  colnames(dat) = c("time","temperature", "mass",
                     "percent",	"deriv.weight")
  
  # source("~/DeterminingScaleFactors.R")
  # 
  # DetermineScale(right=c(min(dat$deriv.weight),
  #                  max(dat$deriv.weight)),
  #                left=c(min(dat$percent),
  #                  max(dat$percent)))
  
  capFig("Example TGA and DTG Curve")
  dat %>%
    mutate(deriv.weight = deriv.weight*84.663+99.502) %>%
    ggplot(aes(x=temperature)) +
    geom_line(aes(y=percent),
              color="black") +
    geom_line(aes(y=deriv.weight),
              color="red") +
    scale_y_continuous(
      # Features of the first axis
      name = "Weight Loss (%)",
      # Add a second axis and specify its features
      sec.axis = sec_axis(trans=~(.-99.502)/84.663,
                          name="DTG (%/°C)")
    ) +
    labs(x= "Temperature (°C)",
         title= "TGA (black) and DTG (red)") +
    theme_bw()
```


In this document, I review the current coral TGA literature, describe the samples I have tested thus far and the proposed analysis methodology, and begin to analyze my initial results.

# Literature Review

There are a handful of publications which have used TGA to analyze coral skeletons grown or collected under acidified environments. The most interesting ones in my opinion are Coronado *et al.* 2019, Moynihan *et al.* 2021, and Prada *et al.* 2021 for their use of the method and the findings they present. A complete list of references I draw upon is included below.

### Coronado *et al.* 2019
This paper sought to disentangle the biological effects on calcification to make corals a more reliable paleoclimate proxy (i.e. better their understand vital effects). They cultured *Stylophora pistillata* under pH 8.2, 7.6, and 7.3 and found differences in the aragonite precipitated by the corals under these treatments due to the increased amount of intracrystalline organic matrix and water content as measured by TGA.

They heated at a rate of 10°C/min to 1000°C, but limited their analysis to discrete regions from ambient to 520°C, focusing mainly on the decomposition of water from ambient to 275°C and bulk organic matrix, OH groups, and possible ACC from 275-520°C (Table 1). There was no clear pattern for water, but for the 275-520°C group there was a lineal increase across treatments(Fig. 2).

```{r Coronado-table}
coronadoTable <- structure(list(Temperature = c("20-150 ", "150-210 ", "210-275 ", 
"275-300 ", "300-330 ", "330-411 ", "411-440 ", "440-520 ", "20-275 ", 
"275-520"), Name = c("Nonstructural water", "Structural Water", "", "", "ACC", 
"", "", "", "H2O", "OM-OH-ACC"), Reasoning = c("evaporation of nonstructural water molecules", 
"", "", "", "crystallization of ACC, which occurs ~ 316", "", 
"transformation of aragonite to calcite, which occurs ~418", 
"", "Integrates over all nonstructural water", "captures all organic matrix, structural water, and ACC crystallization"
)), class = "data.frame", row.names = c(NA, -10L))

coronadoTable %>%
  kbl(align = 'c',
      digits = 4,
      caption = "Table 1: Coronado et al. 2019 analysis Regions") %>%
    kable_classic() %>%
    row_spec(0, bold = T)

```

```{r}
capFig("TGA Results from Coronado et al. 2019")
```
![](/notebook/images/tgaAnalysis/Coronado2019Results.png)<!-- -->

### Moynihan *et al.* 2021
This paper looked at *Porites* coral cores collected from Thailand, Singapore, and Taiwan and correlated micro-mechanical properties with environmental conditions. They found that corals from low salinity and high sedimentation environments (large freshwater influx) had higher organic content and increased embrittlement.  

Their analysis looked at TGA from 20-500°C, which was achieved by heating to a 200C isotherm for 5 minutes followed by a 10°C/min ramp to 400°C and then followed by a 20°C/min ramp to 500°C. They used the DTG curve to identify the temperature of maximum weight loss change (Tmax) and then defined three regions to determine percent change: 175 to 225°C, Tmax-50 - Tmax+50°C, and Tmax+50 to 500°C. They measured micro-mechanical properties with an nanoindentation test.

```{r}
capFig("Regression of micro-mechanical properties and TGA, Moynihan et al. 2019")
```
![](/notebook/images/tgaAnalysis/Moynihan2021.png)<!-- -->


### Prada *et al.* 2021
This paper looked at four populations of naturally occurring reefs in Papa New Guinea, two located by a CO<sub>2</sub> seep and two nearby control sites. They measured 32 samples in total (10 for each control and seep site at Dobu and 6 for each site at Upa Upasina). Their analysis looked at TGA from 20-600°C at 10°C/min incremeents with a 120°C istotherm for 5 minutes to stabalize the removal of adsorbed water. They then focussed their analysis on the region from 125-250°C (structured water molecules) and 250-470°C (organix matrix). 

They only found significant differences between sites in intraskeletal strucutred water and intraskeletal OM at Upa Upasina, suggesting that differences may not be solely environmentally controlled and their may be physiological control of these parameters which are different between different populations. Though further evidence to support this claim is not presented here.

```{r}
capFig("TGA Results from Prada et al. 2021")
```
![](/notebook/images/tgaAnalysis/Prada2021.png)<!-- -->

## Takeaways

Each of these papers analyzed TGA data slightly differently. However, there is general agreement in the organic matrix limits (250-470°C) and the structural water limits (150-225°C). There is potentially other regions of interest too including the crystallization of ACC (300-330°C, ~316°C) and the monotropic transformation of aragonite to calcite (411-440°C, ~418°C). See section below where I propose the analysis I will undertake.


## References
1. Coronado, I., Fine, M., Bosellini, F. R. & Stolarski, J. Impact of ocean acidification on crystallographic vital effect of the coral skeleton. Nat. Commun. 10, 1–9 (2019).
2. Cuif, J. P., Dauphin, Y. Y., Doucet, J., Salome, M. & Susini, J. XANES mapping of organic sulfate in three scleractinian coral skeletons. Geochim. Cosmochim. Acta 67, 75–83 (2003).
3. Cuif, J. P. et al. Fine-scale growth patterns in coral skeletons: Biochemical control over crystallization of aragonite fibres and assessment of early diagenesis. Geol. Soc. Spec. Publ. 303, 87–96 (2008).
4. Cuif, J. P., Dauphin, Y., Berthet, P. & Jegoudez, J. Associated water and organic compounds in coral skeletons: Quantitative thermogravimetry coupled to infrared absorption spectrometry. Geochemistry, Geophys. Geosystems 5, 1–9 (2004).
5. Dauphin, Y., Cuif, J. P. & Massard, P. Persistent organic components in heated coral aragonitic skeletons-Implications for palaeoenvironmental reconstructions. Chem. Geol. 231, 26–37 (2006).
6. Falini, G. et al. Control of aragonite deposition in colonial corals by intra-skeletal macromolecules. J. Struct. Biol. 183, 226–238 (2013).
7. Goffredo, S. et al. Biomineralization control related to population density under ocean acidification. Nat. Clim. Chang. 4, 593–597 (2014).
8. Moynihan, M. A. et al. Environmental impact on the mechanical properties of Porites spp. corals. Coral Reefs 40, 701–717 (2021).
9. Pasquini, L. et al. Isotropic microscale mechanical properties of coral skeletons. J. R. Soc. Interface 12, 1–9 (2015).
10. Prada, F. et al. Coral micro- and macro-morphological skeletal properties in response to life-long acclimatization at CO2 vents in Papua New Guinea. Sci. Rep. 11, 1–10 (2021).
11. Reggi, M. et al. Biomineralization in mediterranean corals: The role of the intraskeletal organic matrix. Cryst. Growth Des. 14, 4310–4320 (2014).
12. Schmidt, M. P., Ilott, A. J., Phillips, B. L. & Reeder, R. J. Structural changes upon dehydration of amorphous calcium carbonate. Cryst. Growth Des. 14, 938–951 (2014).
13. Stolarski, J. & Mazur, M. Nanostructure of biogenic versus abiogenic calcium carbonate crystals. Acta Palaeontol. Pol. 50, 847–865 (2005).

# Samples

The corals for this first batch of analysis come from Allyson Demerlis's recent project where she investigated gene expression during a rapid bleaching experiment of 3 *Acropora cervicornis* genotypes. Some of the corals underwent a thermal stress-hardening treatment where the corals were exposed to a daily variable temperature which is believed to increase a coral's bleaching resilience. The corals exposed to the high temperatures (36°C) had high amounts of mortality and I took these skeletons to analyze. 

```{r allysonStressHardening}
corals <- read_excel('~/Grad School/ULink 2022/AllysonGenotypedCorals/metadata.xlsx') %>%
  mutate(bw_initial_calc = Bw_initial_raw_mass/(1-(rho(S = `bw_initial_salinity (ppt)`,
                                                       T = bw_initial_temp)/1000)/2.93),
         bw_final_calc = bw_final_raw_mass/(1-(rho(S = bw_final_salinity,
                                                   T = bw_final_temp)/1000)/2.93),
         newMassStd = (bw_final_calc-bw_initial_calc)*1000/
           bw_initial_calc/
           as.numeric(bw_final_date-bw_initial_date)) %>%
  filter(Species == 'Acropora cervicornis',
         newMassStd>0) %>%
  filter(ID %in%
           (read.table('~/Grad School/ULink 2022/AllysonGenotypedCorals/skeletonTag.txt') %>%
           pull(V1) %>%
           str_extract(., '[[:digit:]]*') %>%
           as.numeric(.)))

capFig("Calcification data from Allyson Demerlis's project")

corals %>%
  ggplot(aes(Colony,newMassStd, fill=Treatment)) +
  geom_boxplot() +
  theme_classic() +
  labs(y = "Daily Calcification Rate (mg/g/day)",
       x= "Genotype",
       title = "Genotype x Temperature")
```

There was a significant effect of the coral genotype (F=23.27, p=7.78e-6), but no observed effect of stress hardening treatment on calcification rates (F=0.91, p=0.352). Following post-hoc testing, the significant genotype effect was driven by genotype MB-B which had much higher rates than BC-8B and SI-C which had similar calcification rates.

Thus, we are conducting these tests with a special focus on genotype MB-B to see if its faster calcification rate results in any observable differences in skeletal properties.

# Proposed Analysis

With the reviewed literature above, I synthesized the analysis I plan to conduct with the goals of 1) automating the analysis; 2) characterizing the structural water, organic matrix, ACC, aragonite to calcite, and calcium carbonate regions; and 3) analyzing if there are any propertie conserved among genotypes.

## Programatic Analysis

The first way I will analyze will be to find inflection points of the DTG curve to identify the start, peak, and stop of the organc matrix region between 240 and 450°C and the calcium carbonate region between 450 and 850°C. I first calculate the second derivative of the TGA curve and identify all points of inflection (f'' goes from positive to negative). In theory there should only be three points of inflection denoting the start, peak, and stop of the DTG curve. However, due to noise within the data, there is often instances where there is five or more points of inflection. To counteract that, I also run a cumulative minimum function to find the minimum of the DTG curve which corresponds to the peak. I then find the 3 values within the returned point of inflection which center around the peak.

Here is an example. My inflection function returns the values 12,10,7,6,9,10,12. And the cummin function returns the value 6. I center this around the inflection point and extract 7,6, and 9. 

This removes the observed noise and when graphed below, you can see that the extract points correspond with the inflection points you can visualize. 

The other ways I will analyze the data will be over discrete intervals including, 
1. Non structural water (40-150)
2. Structural water (125-250)
3. Total water (40-250)
4. Organic matrix (250-470)
5. Crystallization of ACC (300-330)
6. Aragonite to Calcite Transformation (411-440)
7. Organic Matrix + ACC + OH Groups (275-520)
8. CaCO3 (600-800)

# Initial Results

```{r TGA-Analysis}
source('~/Grad School/ULink 2022/TGA/tgaAnalysis.R')

tga <- list.files(path = '~/Grad School/ULink 2022/TGA/allysonGenotypes/',
           pattern = ".txt",
           full.names = T)

analysis <- lapply(tga, analyze_tga)

lapply(analysis, function(x) {
  x[[1]]
})
TGAdata <- lapply(analysis, function(x) {
        x[[2]]
}) %>% bind_rows()

TGAdata %>%
  relocate(sample, .before=region) %>%
  arrange(region) %>%
   kbl(align = 'c',
      digits = 4,
      caption = "Table 2: Batch 1 Results")


```

# Stats

```{r genotypeStats}
TGAdata <- TGAdata %>%
  mutate(sample=as.numeric(sample)) %>%
  left_join(read_excel('~/Grad School/ULink 2022/AllysonGenotypedCorals/metadata.xlsx') %>%
              filter(Species=='Acropora cervicornis') %>%
              select(ID, genotype=Colony),
            by = c('sample'='ID'))

capFig("Genotype Specific Percent Loss")
TGAdata %>%
  ggplot(aes(genotype, loss)) +
  geom_boxplot() +
  facet_wrap(~region, scales = "free")
```
 The organics figure looks like MB-B has a slight difference, but looking at the y axis its median is only 4% less than the median of the other two genotypes and currently not showing significant genotype differences (F=2.027, p=0.247).

</div>
